common/app/views/support/ImageProfile.scala

```diff
@@ -269,28 +284,26 @@ object ImgSrc extends Logging with implicits.Strings {
     maybePath: Option[String] = None,
     maybeImageMedia: Option[ImageMedia] = None,
     hidpi: Boolean = false
-  ): String = {
+  ): Seq[SrcSet] = {
     val isPng = maybePath.exists(path => path.toLowerCase.endsWith("png"))
     breakpointWidth.toPixels(breakpointWidths)
       .map(browserWidth => ImageProfile(width = Some(browserWidth), hidpi = hidpi, isPng = isPng))
-      .map { profile => {
+      .flatMap { profile => {
         maybePath
           .map(url => srcsetForProfile(profile, url, hidpi))
           .orElse(maybeImageMedia.map(imageContainer => srcsetForProfile(profile, imageContainer, hidpi)))
-          .getOrElse("")
       } }
-      .mkString(", ")
   }
 
   def srcsetForProfile(
     profile: ImageProfile,
     imageContainer: ImageMedia,
     hidpi: Boolean
-  ): String =
-    s"${profile.bestSrcFor(imageContainer).get} ${profile.width.get * (if (hidpi) 2 else 1)}w"
+  ): SrcSet =
+    SrcSet(profile.bestSrcFor(imageContainer).get, profile.width.get * (if (hidpi) 2 else 1))
```

Comment:
`get` is prone to explode if there's a possibility of undefined values.  Would be safer to map over `profile.bestSrcFor(imageContainer)` and `profile.width` and give some default value where one of them isn't defined.

Reply:
yep good shout!